spec/models/service_template_transformation_plan_spec.rb

```diff
@@ -0,0 +1,65 @@
+describe ServiceTemplateTransformationPlan do
+  subject { FactoryGirl.create(:service_template_transformation_plan) }
+
+  describe '#request_class' do
+    it { expect(subject.request_class).to eq(ServiceTemplateTransformationPlanRequest) }
+  end
+
+  describe '#request_type' do
+    it { expect(subject.request_type).to eq("transformation_plan") }
+  end
+
+  describe '#validate_order' do
+    it 'always allows a plan to be ordered' do
+      expect(subject.validate_order).to be_truthy
+    end
+  end
+
+  let(:transformation_mapping) { FactoryGirl.create(:transformation_mapping) }
+  let(:vm1) { FactoryGirl.create(:vm_or_template) }
+  let(:vm2) { FactoryGirl.create(:vm_or_template) }
+
+  let(:catalog_item_options) do
+    {
+      :name        => 'Transformation Plan',
+      :description => 'a description',
+      :config_info => {
+        :transformation_mapping_id => transformation_mapping.id,
+        :vm_ids                    => [vm1.id, vm2.id],
+      }
+    }
+  end
+
+  describe '.create_catalog_item' do
+    it 'creates and returns a transformation plan' do
+      service_template = described_class.create_catalog_item(catalog_item_options)
+      service_template.reload
```

Comment:
Why is this reloaded needed if the call is returning the object for the first time?

If it is needed move the `reload` to the end of the first line.